### ISSUE TITLE
H2 costs are incorrectly calculated

### ISSUE THREAD

**Issue by robbieorvis** [Energy-Innovation/eps-us#326]
H2 costs, particularly for electrolytic H2, are not calculated correctly when allowing the model to adjust prices based on production costs. We either need to have H2 costs fully endogenized or move to a new system. In addition to the current issues, the H2 sector also uses industrial rates for electricity for H2, but facilities are contemplating contracting directly with wholesale providers, often onsite, and their electricity purchase costs would therefore be MUCH lower than the retail rate. For now we have disabled the model's adjustment of prices based on production costs. Needs to be wholly revisited in the future.

**Comment by dobrien13** [comment]
When we get to this, it'd be worth exploring how to pass changes in H2 production costs through to consumers (like we do with electricity). Presently, adding a carbon tax on fuels (NG) being consumed to make H2 does not increase the cost of H2 consumed in industry.

**Comment by dobrien13** [comment]
We are now calculating H2 costs endogenously as of 4.1. However, we still use industrial electricity prices for power. Eric offered the following. Let me know what you think, @robbieorvis, if you want me to shift to pull in wholesale or stick to industrial. Simple fix now. 

> It depends.  In a state like CA the don't have access to nodal and/or wholesale prices, while in Texas they do.  There is also a lot more dispersion in the distribution of electricity rates around the state average than for residential or commercial rate payers.  Finally, large industrial users almost always negotiate some kind of sweetheart deal for electricity (similarly for gas) by negotiating bilaterally with utilities and PUCs — almost always non-transparent.  
> 
> And yet, even with all these advantages,  a lot of not most of large industrials tend to have onsite generation for lower costs, improved efficiency (CHP) and resilience purposes (can't afford loss of access to the grid).  
> 
> Don't know how much that helps :). They will pay somewhere between wholesale and industry rates, but in the best cases the will pay even less by consuming flexibly.  E.g Talus has access to 4c per kWh power from midwestern utilities but being a controllable load with ~55% load factor.

**Comment by robbieorvis** [comment]
On a state-by-state basis, SEDS reflects this variation. For example the electricity price in industry is 17 in Texas but 63 in California.

When we compute the electricity costs in the model we look at the difference between the models' computed price and the SEDS price and that serves as our starting adder, which can be negative, that we add to the computed wholesale energy and T&D cost computed in the model. So I believe we are capturing this dynamic just fine, of course noting that the more highly aggregated a region is, the more the weighted average nets out these nuances.

In short, I don't think there's anything to do for now on this front, at least not unless/until we have subregional modeling.
